src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,37 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { Todo } from '../../types/Todo';
 import { Loader } from '../Loader';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+
+type Props = {
+  selectedTodo: Todo | null;
```

Comment:
```suggestion
  selectedTodo: Todo;
```
You render modal only when you have selected todo, so it could be required prop